torchvision/transforms/v2/functional/_misc.py

```diff
@@ -340,6 +347,90 @@ def _to_dtype_tensor_dispatch(inpt: torch.Tensor, dtype: torch.dtype, scale: boo
     return inpt.to(dtype)
 
 
+# cvcuda is only used if it is installed, so we can simply define empty mappings
+_torch_to_cvcuda_dtypes = {}
+_cvcuda_to_torch_dtypes = {}
+if CVCUDA_AVAILABLE:
+    # put the entire conversion set here
+    # only a subset are used for torchvision
+    _torch_to_cvcuda_dtypes = {
+        torch.uint8: cvcuda.Type.U8,
+        torch.uint16: cvcuda.Type.U16,
+        torch.uint32: cvcuda.Type.U32,
+        torch.uint64: cvcuda.Type.U64,
+        torch.int8: cvcuda.Type.S8,
+        torch.int16: cvcuda.Type.S16,
+        torch.int32: cvcuda.Type.S32,
+        torch.int64: cvcuda.Type.S64,
+        torch.float32: cvcuda.Type.F32,
+        torch.float64: cvcuda.Type.F64,
+    }
+    # create reverse mapping
+    _cvcuda_to_torch_dtypes = {v: k for k, v in _torch_to_cvcuda_dtypes.items()}
+
+
+def _to_dtype_cvcuda(
+    inpt: "cvcuda.Tensor",
+    dtype: torch.dtype,
+    scale: bool = False,
+) -> "cvcuda.Tensor":
+    """
+    Convert the dtype of a CV-CUDA tensor, based on a torch.dtype.
+
+    Args:
+        inpt: The CV-CUDA tensor to convert the dtype of.
+        dtype: The torch.dtype to convert the dtype to.
+        scale: Whether to scale the values to the new dtype.
+            There are four cases for the scaling setup:
+            1. float -> float
+            2. int -> int
+            3. float -> int
+            4. int -> float
+            If scale is True, the values will be scaled to the new dtype.
+            If scale is False, the values will not be scaled.
+            The scale values for float -> float and int -> int are 1.0 and 0.0 respectively.
+            The scale values for float -> int and int -> float are the maximum value of the new dtype.
+
+    Returns:
+        out (cvcuda.Tensor): The CV-CUDA tensor with the converted dtype.
+
+    """
+    cvcuda = _import_cvcuda()
+
+    dtype_in = _cvcuda_to_torch_dtypes[inpt.dtype]
+    cvc_dtype = _torch_to_cvcuda_dtypes[dtype]
+
+    if not scale:
+        return cvcuda.convertto(inpt, dtype=cvc_dtype)
```

Comment:
We could maybe avoid having this early return statement I feel? The default values for `cvcuda.convertto` are `scale: float = 1, offset: float = 0` according to the [documentation](https://cvcuda.github.io/CV-CUDA/modules/python/operators.html#cvcuda.convertto). We could just make sure that `scale_val, offset = 1.0, 0.0` when `scale` is `None` and return at the end of the function?